### PR TITLE
Fix: EnsemblVEP parsing issues, missing info in VCF entries, timeline and report overwriting

### PR DESCRIPTION
**Pull request by groodri**
- Enable overwriting of nextflow timeline and report; cleanup nextflow work directory on successful workflow completion
- Hotfix for missing info tab in VCF entries for SNPs in ends of amplicons with some read depth but zero alternative nucleotide counts
- Hotfix for EnsemblVEP API responses that caused a JSON decoding error
- Small refactoring of variable names associated with variant calling code; make edge of amplicon detection a user argument if `detect_indel.py` is main

### CODE REVIEW

File: bin/variant_calling/detect_indel.py [hunk 1]
```diff
@@ -240,13 +242,16 @@ def extract_indel_evidence(
...
-                ]
-                ref_seq = "".join(ref_seq)
-                alleles = (str(ref_seq).upper(), str(ref_seq[0]).upper())
+                if pos + indel.length + 1 <= amplicon_end:
+                    # Adjust reference allele to include deleted seq
+                    ref_seq = [
+                        reference_mapper[x].upper()
+                        for x in range(pos, pos + indel.length + 1)
+                    ]
+                    ref_seq = "".join(ref_seq)
```
Comment (da-i) [inline bin/variant_calling/detect_indel.py:251]: upper here

File: bin/variant_calling/detect_indel.py [hunk 2]
```diff
@@ -240,13 +242,16 @@ def extract_indel_evidence(
...
+                    alleles = (str(ref_seq).upper(), str(ref_seq[0]).upper())
```
Comment (da-i) [inline bin/variant_calling/detect_indel.py:252]: no str needed

File: bin/variant_calling/detect_indel.py [hunk 3]
```diff
@@ -324,74 +329,84 @@ def main(
...
+        contig_region_stop = int(bed_file_line[2])
+        contig_reference_mapper = create_ref_mapper(
+            fasta, contig, contig_region_start, contig_region_stop
         )
 
-        ref_seq = reference.fetch(reference=contig, start=pos, end=pos + 1)
+        for pos in range(contig_region_start, contig_region_stop):
+            close_to_edge = (
+                # close to start
+                pos - 4 <= int(contig_region_start)
```
Comment (da-i) [inline bin/variant_calling/detect_indel.py:351]: why is it 4?

File: bin/variant_calling/detect_indel.py [hunk 4]
```diff
@@ -324,74 +329,84 @@ def main(
...
-                for tag_id, value in result[3].items():
-                    r.info[tag_id] = value
-
-                vcf.write(r)
-
-            else:
-                # Reference allele is '.', then no variant was found
-                # Don't write anthing to VCF output file
-                continue
+                if result[1][0] != ".":
```
Comment (da-i) [inline bin/variant_calling/detect_indel.py:380]: unpack and name before using it

File: nextflow.config [hunk 5]
```diff
@@ -214,10 +214,12 @@ params{
...
   enabled = true
+  overwrite = true
   file = "${params.output_dir}/execution/nextflow_timeline.html"
 }
 report {
   enabled = true
+  overwrite = true
   file = "${params.output_dir}/execution/nextflow_report.html"
 }
 trace {
```
Comment (da-i) [inline nextflow.config:225]: cleanup??

File: bin/variant_calling/detect_indel.py (hunk 3 above)
Comment (groodri) [reply bin/variant_calling/detect_indel.py:351]: made into variable `min_edge_distance: int = 4`

**Comment by groodri** [comment]
Successful test on CYC000432

**Pull request merged**